src/Plugin/Block/OpenseadragonBlock.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Drupal\openseadragon\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Provides a 'OpenseadragonBlock' block from a IIIF manifest.
+ *
+ * @Block(
+ *  id = "openseadragon_block",
+ *  admin_label = @Translation("Openseadragon block"),
+ * )
+ */
+class OpenseadragonBlock extends BlockBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [] + parent::defaultConfiguration();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state) {
+    $form['iiif_manifest_url_fieldset'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('IIIF Manifest URL'),
+    ];
+    $form['iiif_manifest_url_fieldset']['iiif_manifest_url'] = [
+      '#type' => 'textfield',
+      '#description' => $this->t('Absolute URL of the IIIF manifest to render.  You may use tokens to provide a pattern (e.g. "http://localhost/node/[node:nid]/manifest")'),
+      '#default_value' => $this->configuration['iiif_manifest_url'],
+      '#maxlength' => 256,
+      '#size' => 64,
+      '#required' => TRUE,
+    ];
+    $form['iiif_manifest_url_fieldset']['token_help'] = [
+      '#theme' => 'token_tree_link',
+      '#global_types' => FALSE,
+      '#token_types' => ['node'],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSubmit($form, FormStateInterface $form_state) {
+    $this->configuration['iiif_manifest_url'] = $form_state->getValues()['iiif_manifest_url_fieldset']['iiif_manifest_url'];
```

Comment:
Seemingly the correct form of this is
```
$form_state->getValue(['iiif_manifest_url_fieldset', 'iiif_manifest_url']);
```
As it uses [NestedArray::getValue](https://api.drupal.org/api/drupal/core!lib!Drupal!Component!Utility!NestedArray.php/function/NestedArray%3A%3AgetValue/8.7.x)

Finding the syntax was a little bit of duckduckgo'ing

Reply:
wow... i had no idea you could do that...

I'll update it.  Also, found that there's a validation function i can toss in, too.